src/library_websocket.js

```diff
@@ -294,7 +294,7 @@ var LibraryWebSocket = {
     var url = UTF8ToString({{{ makeGetValue('createAttributes', 0, '*') }}});
     var protocols = {{{ makeGetValue('createAttributes', C_STRUCTS.EmscriptenWebSocketCreateAttributes.protocols, '*') }}}
     // TODO: Add support for createOnMainThread==false; currently all WebSocket connections are created on the main thread.
-    // var createOnMainThread = HEAP32[createAttrs+2];
+    // var createOnMainThread = HEAP8[createAttributes+2];
```

Comment:
This looks wrong. That's a pointer.

Reply:
It looks right to me.   createAttributes is a pointer a struct.  Two bytes into that struct is a boolean called `createOnMainThread`: https://github.com/emscripten-core/emscripten/blob/b8c6e231ee05cbadef1abd9e41dde5cd19e5a733/system/include/emscripten/websocket.h#L96

Reply:
Ah, I see, thanks.

Reply:
(But it was wrong before, then: the HEAP32 was missing the `>> 2`. Anyhow, it's in a comment...)